pimd/pim_igmp.h

```diff
@@ -127,6 +127,10 @@ static inline void pim_igmp_other_querier_timer_on(struct gm_sock *igmp)
 static inline void pim_igmp_other_querier_timer_off(struct gm_sock *igmp)
 {
 }
+
+static inline void pim_igmp_if_reset(struct pim_interface *pim_ifp)
```

Comment:
In case of pimv6 here, call gm_group_delete function inside here. So we can remove the if else check in the caller. Change the api name to pim_gm_if_reset as well.